MoTrPAC/exercise-transcriptome-meta-analysis
# Connects to #1. Connects to #3. Connects to #6. Connects to #7. Connects to #8. Connects to #9.

### Key Changes

1. Implemented **Autosuggest** support for gene search input.
2. Added 4 more columns to the gene meta-analysis input table: **%Males, Sample Size, Time, and Beta**.
3. Updated the **README.md** file.
4. Implemented support to handle various use cases of displaying values in the tables:
   * Rounding `4.34225220539258e-09` to `4.34e-9`
   * Rounding `0.0034225220539258` to `3.42e-3`
   * Displaying a `0` value as `0`
   * Displaying a `null` value as `Unavailable`
5. David Amar provided revised gene stats data in `*.tsv` files.
6. Minor UI changes to the plot image due to the addition of 2 footnotes regarding the plot.
7. Dropped custom `font-family` in favor of `Bootstrap`'s default native font stack.

### Technical Notes

1. Reviewing this PR requires `node.js v14.15.4`.
2. Start the site by running `yarn start` after running `yarn install`.

### Steps to test

1. On the landing page, enter `veg` in the text input field and expect to see a list of suggested gene symbols.
2. Select a `VEGFB` and press the <kbd>return</kbd> key. Expect to see the UIs being changed to show the searched gene's name and summary.
3. Click on the Acute Muscle (or Longterm Muscle) tab and expect to see the new **%Males, Sample Size, Time, and Beta** columns present in the analysis input table (2nd table).
4. Also expect to see numbers with exponential notations have no more than 2 decimals.
5. Lastly, expect to see 2 footnote items being shown at the bottom of the page when the forest plot images are present.
